treeherder/webapp/api/push.py

```diff
@@ -42,7 +45,7 @@ def list(self, request, project):
 
         # This will contain some meta data about the request and results
         meta = {}
-
+        search_query = filter_params.get("q")
```

Comment:
move this request just before the related `if` below.

Also as a matter of readability, I'd do these extra changes:
* rename the "q" parameter to "search"
* rename "search_query" to "search_param"